File: src/main/java/dev/rafi/vitelist/commands/sub/Add.java
package dev.rafi.vitelist.commands.sub;

import com.velocitypowered.api.command.CommandSource;
import dev.rafi.vitelist.storage.Config;
import dev.rafi.vitelist.utils.ChatUtils;
import dev.rafi.vitelist.utils.QueryUtils;

public class Add {
    public static void whitelistAdd(CommandSource executor, String[] args) {
        if (!executor.hasPermission("vitelist.commands.add")) {
            ChatUtils.sendMessage(executor, Config.NO_PERMISSION);
            return;
        }

        if (args.length == 1) {
            ChatUtils.sendMessage(executor, Config.WRONG_USAGE);
        } else if (args.length == 2) {
            QueryUtils.addToWhiteList(args[1], "all");
            ChatUtils.sendMessage(executor, Config.ADD_GLOBAL.replace("%user%", args[1]));
        } else if (args.length == 3) {
            if (!QueryUtils.serverExists(args[2])) {
                ChatUtils.sendMessage(executor, Config.WRONG_SERVER.replace("%server%", args[2]));
                return;
            }
            QueryUtils.addToWhiteList(args[1], args[2]);
            ChatUtils.sendMessage(executor, Config.ADD_SERVER.replace("%user%", args[1]).replace("%server%", args[2]));
        }
    }
}

File: src/main/kotlin/dev/rafi/vitelist/commands/subcommands/Add.kt
package dev.rafi.vitelist.commands.subcommands

import com.velocitypowered.api.command.CommandSource
import dev.rafi.vitelist.database.model.Server
import dev.rafi.vitelist.storage.Config
import dev.rafi.vitelist.utils.sendMessage
import dev.rafi.vitelist.utils.serverExists
import org.jetbrains.exposed.sql.SqlExpressionBuilder.plus
import org.jetbrains.exposed.sql.selectAll
import org.jetbrains.exposed.sql.transactions.transaction
import org.jetbrains.exposed.sql.update

fun add(source: CommandSource, args: Array<String>) {
    if (!(source.hasPermission("vitelist.commands.add"))) {
        sendMessage(source, Config.NO_PERMISSION)
        return
    }
    if (args.size == 1) {
        sendMessage(source, Config.WRONG_USAGE)
    } else if (args.size == 2) {
        val playerName = args[1]

        transaction {
            Server.selectAll().forEach { server ->
                if (server[Server.players].contains(playerName)) return@forEach

                Server.update({ Server.id eq server[Server.id] }) {updateServer ->
                    updateServer[players] = players.plus("$playerName,")
                }
            }
        }
        sendMessage(source, Config.ADD_GLOBAL.replace("%user%", playerName))
    } else {
        val playerName = args[1]
        val serverName = args[2]

        if (!serverExists(serverName)) {
            sendMessage(source, Config.SERVER_NOT_FOUND.replace("%server%", serverName))
            return
        }

        transaction {
            Server.select(Server.name, Server.players).where {Server.name eq serverName}.forEach { server ->
                if (server[Server.players].contains(playerName)) return@forEach

                Server.update({ Server.name eq server[Server.name] }) { updateServer ->
                    updateServer[players] = players.plus("$playerName,")
                }
            }
        }
        sendMessage(source, Config.ADD_SERVER.replace("%user%", playerName).replace("%server%", serverName))
    }
}